taskrun: include actual result size in error when exceeding maxResultSize

a-ateek opened the pull request:
Fixes #8868

# Changes

Previously, when a TaskRun failed due to a result exceeding the maxResultSize, the error message did not include the actual size of the offending result. This makes it harder for users to figure out how much they exceeded the configured limit.

Example: 
```
  Conditions:
    Last Transition Time:  2025-07-03T17:59:57Z
    Message:               [User error]  TaskRun ""max-size-run"" failed: results exceeded size limit 20 bytes
    Reason:                TaskRunResultLargerThanAllowedLimit
```

This change improves the error message to include the actual size to debug pipelines with large results.
Example: 
```
  Conditions:
    Last Transition Time:  2025-07-05T12:08:56Z
    Message:               [User error]  TaskRun ""max-size-run"" failed: 47 bytes results size exceeds configured limit of 20 bytes
    Reason:                TaskRunResultLargerThanAllowedLimit
```

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [pre-commit](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#install-tools) Passed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Improved error message when a TaskRun result exceeds `maxResultSize` to include the actual result size, making it easier to debug oversized results
```

a-ateek commented:
/kind feature

pritidesai commented:
Thank you @a-ateek for this PR. Please consider adding unit tests to cover the proposed changes.

a-ateek commented:
@pritidesai Unit tests already existed for this function and now I have adjusted them according to new error message. Thank you for pointing that out. Also rebased the branch.

waveywaves commented:
/retest

waveywaves commented:
@a-ateek  would you mind squashing your commits, rebasing and pushing your branch again ?

waveywaves commented:
@a-ateek would you mind squashing your commits, rebasing and pushing your branch again ? we have recently removed some gittea tests which might be causing these test failures. Just to eliminate the possiblility a rebase would be great.

vdemeester commented:
/retest

vdemeester commented:
/lgtm
